modules/reactor/src/meshgenerators/FlexiblePatternGenerator.C

```diff
@@ -0,0 +1,454 @@
+      "extra_positions_mg_indices",
+      "the indices of the input mesh generators for the extra position.");
+
+  params.addParam<std::vector<std::vector<std::vector<unsigned int>>>>("hex_patterns",
+                                                                       "Hexagonal patterns set.");
+  params.addRangeCheckedParam<std::vector<Real>>(
+      "hex_pitches", "hex_pitches>0", "pitch sizes used to generate the hexagonal patterns.");
+  params.addParam<std::vector<Point>>("hex_origins",
+                                      "the origin positions of the hexagonal patterns,");
+  params.addParam<std::vector<Real>>("hex_rotations",
+                                     "the rotation angles of the hexagonal patterns,");
+
+  params.addParam<std::vector<std::vector<std::vector<unsigned int>>>>("rect_patterns",
+                                                                       "Rectangular patterns set.");
+  params.addRangeCheckedParam<std::vector<Real>>(
+      "rect_pitches_x",
+      "rect_pitches_x>0",
+      "pitch sizes in x direction used to generate the rectangular patterns.");
+  params.addRangeCheckedParam<std::vector<Real>>(
+      "rect_pitches_y",
+      "rect_pitches_y>0",
+      "pitch sizes in y direction used to generate the rectangular patterns.");
+  params.addParam<std::vector<Point>>("rect_origins",
+                                      "the origin positions of the rectangular patterns,");
+  params.addParam<std::vector<Real>>("rect_rotations",
+                                     "the rotation angles of the rectangular patterns.");
+
+  params.addParam<std::vector<std::vector<unsigned int>>>("circular_patterns",
+                                                          "Circular patterns set.");
+  params.addRangeCheckedParam<std::vector<Real>>(
+      "circular_radii", "circular_radii>0", "the radii of the circular patterns.");
+  params.addParam<std::vector<Point>>("circular_origins",
+                                      "the origin positions of the circular patterns,");
+  params.addParam<std::vector<Real>>(
+      "circular_rotations",
+      "the rotation angles of the circular patterns (the azimuthal angle of the first unit mesh).");
+
+  // Parameters directly passed to XYDelauanyMeshGenerator
+  params.addParam<Real>("desired_area", 0.0, "Desired are for the background area meshing.");
+  params.addParam<std::string>(
+      "desired_area_func",
+      std::string(),
+      "Desired area as a function of x,y; omit to skip non-uniform refinement");
+  params.addParam<bool>("verify_holes", true, "Whether the holes are verified.");
+
+  params.addParam<boundary_id_type>(
+      "external_boundary_id",
+      "The boundary id of the external boundary in addition to the default 10000.");
+
+  params.addClassDescription("This FlexiblePatternGenerator object is designed to generate a "
+                             "complex mesh with a background region with dispersed unit meshes in "
```

Comment:
```suggestion
                             "mesh with a background region with dispersed unit meshes in "
```
I agree it s complex but you made it easy